### PR TITLE
AccessValidator_RemoveIBus

### PR DESCRIPTION
**Pull request by djino97**
- Reworked AccessValidator methods and reworked requestClient
- Added new nuget version 1.0.5

-  For adding AccessValidator  in services, you need add two methods in Startup: first AddKernelExtensions, then ConfigureKernelMassTransit (in masstransit configurations)

### CODE REVIEW

File: src/Kernel/AccessValidator/Requests/IAccessValidatorCRServiceRequest.cs [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-    {
-        Guid UserId { get; set; }
-        int RightId { get; set; }
-    }
-}
+﻿using System;
+
+namespace LT.DigitalOffice.Kernel.AccessValidator.Requests
+{
+    // <summary>
```
Comment (KostinP) [inline src/Kernel/AccessValidator/Requests/IAccessValidatorCRServiceRequest.cs]: ///

File: test/Kernel.UnitTests/AccessValidator/AccessValidatorTests.cs [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+using MassTransit;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Primitives;
+using Moq;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Threading.Tasks;
+using AV = LT.DigitalOffice.Kernel.AccessValidator;
```
Comment (KostinP) [inline test/Kernel.UnitTests/AccessValidator/AccessValidatorTests.cs:13]: You can just write `using LT.DigitalOffice.Kernel.AccessValidator`

File: src/Kernel/AccessValidator/Requests/IAccessValidatorCRServiceRequest.cs [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
+    /// Message request model that is sent to CheckRightsService via MassTransit.
+    /// </summary>
+    public interface IAccessValidatorCRServiceRequest
```
Comment (KostinP) [inline src/Kernel/AccessValidator/Requests/IAccessValidatorCRServiceRequest.cs]: Perhaps it is worth writing a complete transcript.

File: test/Kernel.UnitTests/AccessValidator/AccessValidatorTests.cs [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+    public class OperationResult<T> : IOperationResult<T>
+    {
+        public bool IsSuccess { get; set; }
+
+        public List<string> Errors { get; set; }
+
+        public T Body { get; set; }
+    }
+
+
```
Comment (KostinP) [inline test/Kernel.UnitTests/AccessValidator/AccessValidatorTests.cs]: Extra empty line.

File: src/Kernel/DependencyInjection/ServiceCollectionExtension.cs [hunk 5]
```diff
@@ -11,9 +14,29 @@ public static IServiceCollection AddKernelExtensions(this IServiceCollection ser
...
+                x.AddRequestClient<IAccessValidatorCRServiceRequest>(
+                    new Uri(options.AccessValidatorCheckRightsServiceURL));
+            });
+
+            services.AddMassTransitHostedService();
+
+            return services;
+        }
+
+    }
```
Comment (KostinP) [inline src/Kernel/DependencyInjection/ServiceCollectionExtension.cs]: Extra empty line.

File: test/Kernel.UnitTests/AccessValidator/AccessValidatorTests.cs [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+        [Test]
+        public void ShouldThrowFormatExceptionWhenThereIsInvalidGuidInHeaders()
+        {
+            httpContextMock
+                .Setup(h => h.HttpContext.Request.Headers["UserId"])
+                .Returns("SampleText");
+
+                Assert.Throws<FormatException>(() => accessValidator.IsAdmin());
+                Assert.Throws<FormatException>(() => accessValidator.HasRights(RIGHT_ID));
+        }
```
Comment (KostinP) [inline test/Kernel.UnitTests/AccessValidator/AccessValidatorTests.cs:191]: Extra spaces.

File: test/Kernel.UnitTests/AccessValidator/AccessValidatorTests.cs [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+
+        [Test]
+        public void ShouldReturnFalseWhenUserDoesntHaveRights()
+        {
+            operationResult.IsSuccess = true;
+            operationResult.Body = false;
+
+            httpContextMock
+                .Setup(h => h.HttpContext.Request.Headers["UserId"])
+                .Returns(userId);
```
Comment (KostinP) [inline test/Kernel.UnitTests/AccessValidator/AccessValidatorTests.cs:158]: Cool.

**Review by KostinP — APPROVED** [review]

File: test/Kernel.UnitTests/AccessValidator/AccessValidatorTests.cs (hunk 2 above)
Comment (djino97) [reply test/Kernel.UnitTests/AccessValidator/AccessValidatorTests.cs:13]: not work

File: src/Kernel/AccessValidator/Requests/IAccessValidatorCRServiceRequest.cs (hunk 1 above)
Comment (djino97) [reply src/Kernel/AccessValidator/Requests/IAccessValidatorCRServiceRequest.cs]: fixed

File: test/Kernel.UnitTests/AccessValidator/AccessValidatorTests.cs (hunk 4 above)
Comment (djino97) [reply test/Kernel.UnitTests/AccessValidator/AccessValidatorTests.cs]: fixed

File: src/Kernel/DependencyInjection/ServiceCollectionExtension.cs (hunk 5 above)
Comment (djino97) [reply src/Kernel/DependencyInjection/ServiceCollectionExtension.cs]: fixed

File: test/Kernel.UnitTests/AccessValidator/AccessValidatorTests.cs (hunk 7 above)
Comment (djino97) [reply test/Kernel.UnitTests/AccessValidator/AccessValidatorTests.cs:158]: What? I do not understand claim

File: test/Kernel.UnitTests/AccessValidator/AccessValidatorTests.cs (hunk 6 above)
Comment (djino97) [reply test/Kernel.UnitTests/AccessValidator/AccessValidatorTests.cs:191]: Where?

File: src/Kernel/AccessValidator/Requests/IAccessValidatorCRServiceRequest.cs (hunk 3 above)
Comment (djino97) [reply src/Kernel/AccessValidator/Requests/IAccessValidatorCRServiceRequest.cs]: There is code documentation

**Review by DarkFelas — APPROVED** [review]

**Comment by KostinP** [comment]
Update version to 1.0.5 :D

File: test/Kernel.UnitTests/AccessValidator/AccessValidatorTests.cs (hunk 7 above)
Comment (KostinP) [reply test/Kernel.UnitTests/AccessValidator/AccessValidatorTests.cs:158]: This is not a claim

File: src/Kernel/AccessValidator/Requests/IAccessValidatorCRServiceRequest.cs (hunk 3 above)
Comment (KostinP) [reply src/Kernel/AccessValidator/Requests/IAccessValidatorCRServiceRequest.cs]: = IAccessValidatorCRServiceRequest -> IAccessValidatorCheckRightsServiceRequest

Comment (djino97) [reply src/Kernel/AccessValidator/Requests/IAccessValidatorCRServiceRequest.cs]: Fixed

**Pull request merged**